html/browsers/windows/clear-window-name.https.html

```diff
@@ -18,91 +18,82 @@
   hyperlink.click();
 }
 
-function pollResultAndCheck(t, id, expected) {
-  return new Promise(resolve => {
-    const stashURL = new URL("resources/window-name-stash.py", location);
-    stashURL.searchParams.set('id', id);
-
-    let checkResult = t.step_func(async () => {
-      let response = await fetch(stashURL);
-      let res = await response.text();
-
-
-      if (res !== "NONE") {
-        if (res !== expected) {
-          t.step(() => assert_unreached('Stash result does not equal expected result.'));
-        }
-        t.done();
-        resolve();
-      } else {
-        t.step_timeout(checkResult, 100);
-      }
-    });
-
-    t.step_timeout(checkResult, 100);
-  });
+async function pollResultAndCheck(t, id, expected) {
+  const stashURL = new URL("resources/window-name-stash.py", location);
+  stashURL.searchParams.set('id', id);
+
+  let res = "NONE";
+  while (res == "NONE") {
+    await new Promise(resolve => { t.step_timeout(resolve, 100); });
```

Comment:
Doesn't matter much, but because invoking `resolve()` can't throw an exception, just `step_timeout` would do, and you could avoid passing in `t` entirely.

Reply:
Fair, but global step_timeout is really meant for single-page tests - let's do the right thing even if it takes an extra character.